core/io/ip.cpp

```diff
@@ -146,31 +155,39 @@ IP::ResolverID IP::resolve_hostname_queue_item(const String &p_hostname, IP::Typ
 			resolver->resolve_queues();
 	}
 
+	resolver->mutex->unlock();
 	return id;
 }
 
 IP::ResolverStatus IP::get_resolve_item_status(ResolverID p_id) const {
 
 	ERR_FAIL_INDEX_V(p_id, IP::RESOLVER_MAX_QUERIES, IP::RESOLVER_STATUS_NONE);
 
-	GLOBAL_LOCK_FUNCTION;
+	resolver->mutex->lock();
 	ERR_FAIL_COND_V(resolver->queue[p_id].status == IP::RESOLVER_STATUS_NONE, IP::RESOLVER_STATUS_NONE);
+	IP::ResolverStatus res = resolver->queue[p_id].status;
 
-	return resolver->queue[p_id].status;
+	resolver->mutex->unlock();
+	return res;
 }
+
 IP_Address IP::get_resolve_item_address(ResolverID p_id) const {
 
 	ERR_FAIL_INDEX_V(p_id, IP::RESOLVER_MAX_QUERIES, IP_Address());
 
-	GLOBAL_LOCK_FUNCTION;
+	resolver->mutex->lock();
 
 	if (resolver->queue[p_id].status != IP::RESOLVER_STATUS_DONE) {
 		ERR_EXPLAIN("Resolve of '" + resolver->queue[p_id].hostname + "'' didn't complete yet.");
 		ERR_FAIL_COND_V(resolver->queue[p_id].status != IP::RESOLVER_STATUS_DONE, IP_Address());
 	}
 
-	return resolver->queue[p_id].response;
+	IP_Address res = resolver->queue[p_id].response;
+
+	resolver->mutex->unlock();
+	return res;
 }
+
 void IP::erase_resolve_item(ResolverID p_id) {
```

Comment:
Still uses `GLOBAL_LOCK_FUNCTION;` might cause race condition, should convert to mutex